internal/limitopen/limitopen.go

```diff
@@ -0,0 +1,104 @@
+package limitopen
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+// Open opens a path for read.
+//
+// It's similar to os.Open, but with the following differences:
+//
+// 1. It returns io.ReadCloser other than *os.File,
+//
+// 2. It returns the size reported by the system to the user.
+//
+// 3. The io.ReadCloser returned would be guaranteed to be not able to read
+//    beyond the size returned (for example, if you use this function to open
+//    /dev/zero, the system would return 0 as the size, and as a result when
+//    reading from r you would get EOF immediately, instead of being able to
+//    read from it indefinitely)
+//
+// It would never return both non-nil r and err.
+// When err is nil it's the caller's responsibility to close r returned.
+func Open(path string) (r io.ReadCloser, size int64, err error) {
+	var f *os.File
+	f, err = os.Open(path)
```

Comment:
you can use `f, err :=` now and don't need the var

Reply:
the named returns are for documentation purpose (if we just make the function signature `(io.ReadCloser, int64, error)` it won't be obvious what the `int64` return is for).